packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -0,0 +1,454 @@
+   * <br />
+   * `inputValue` will be an empty string when no search text is present.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  noMenuOptionsMessage?: ReactSelectProps['noOptionsMessage'];
+  /**
+   * Whether to enable search functionality.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  isMenuSearchable?: ReactSelectProps['isSearchable'];
+  /**
+   * Maximum height of the menu before scrolling
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  maxMenuHeight?: ReactSelectProps['maxMenuHeight'];
+  /**
+   * Whether the menu should close after a value is selected. Defaults to `true`.
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  closeMenuOnSelect?: ReactSelectProps['closeMenuOnSelect'];
+  /**
+   * Horizontal size limit for the dropdown menu.
+   */
+  menuHorizontalConstraint?: 3 | 4 | 5;
+  /**
+   * Show submit button in the input
+   */
+  showSubmitButton?: boolean;
+};
+
+const defaultProps: Pick<
+  TSelectableSearchInputProps,
+  | 'horizontalConstraint'
+  | 'isClearable'
+  | 'menuHorizontalConstraint'
+  | 'showSubmitButton'
+> = {
+  horizontalConstraint: 'scale',
+  isClearable: true,
+  menuHorizontalConstraint: 3,
+  showSubmitButton: true,
+};
+
+const selectableSearchInputSequentialId = createSequentialId(
+  'selectable-search-input-'
+);
+
+const SelectableSearchInput = (props: TSelectableSearchInputProps) => {
+  const [textInputHasFocus, toggleTextInputHasFocus] = useToggleState(false);
+  const [dropdownHasFocus, toggleDropdownHasFocus] = useToggleState(false);
+  const [searchValue, setSearchValue] = useState(props.value.text || '');
+  const containerRef = useRef<HTMLDivElement>(null);
+  const textInputRef = useRef<HTMLInputElement>(null);
+
+  const { isNewTheme } = useTheme();
+
+  const optionsWithoutGroups = flatMap(props.options, (option) =>
+    has(option, 'value') ? option : (option as TOptionObject).options
+  );
+
+  const selectedOption = optionsWithoutGroups.find(
+    (option) =>
+      has(option, 'value') && (option as TOption).value === props.value.option
+  ) as TOption;
+
+  const selectablSearchInputId = useFieldId(
+    props.id,
+    selectableSearchInputSequentialId
+  );
+
+  if (!props.isReadOnly) {
+    warning(
+      typeof props.onChange === 'function',
+      'TextInput: `onChange` is required when is not read only.'
+    );
+  }
+
+  const handleTextInputFocus = useCallback(() => {
+    if (props.onFocus) {
+      props.onFocus({
+        target: {
+          id: SelectableSearchInput.getTextInputId(selectablSearchInputId),
+          name: getTextInputName(props.name),
+        },
+      });
+    }
+    toggleTextInputHasFocus(true);
+  }, [textInputHasFocus, props.onFocus, selectablSearchInputId, props.name]);
```

Comment:
There are some eslint warnings about `props` and `toggleTextInputHasFocus` not being included in the dependencies array (not only for this `useCallback`).
We might need to destructure properties such `onFocus`, `onBlur`,...) to not include the whole `props` object

Reply:
Thanks. I didn't have the eslint extension on the new laptop, so I couldn't see the warnings. 
Updated [cd72889](https://github.com/commercetools/ui-kit/pull/2426/commits/cd72889b1c645bc9d296d204c3808bf5f9450647)